src-tauri/src/cli/helpers.rs

```diff
@@ -140,3 +144,34 @@ pub async fn track_cli_event(event_name: &str, additional_data: Option<serde_jso
       "additional_data": additional_data
     })).await;
 }
+
+/// Sort mirrors by measured latency (ascending), using None for timeouts.
+pub async fn sorted_mirror_entries(mirrors: &[&str]) -> Vec<MirrorEntry> {
+    let latency_map = calculate_mirror_latency_map(&mirrors.to_vec()).await;
+    let mut entries: Vec<MirrorEntry> = Vec::new();
+    for (key, value) in latency_map.iter() {
+        entries.push((key.clone(), value.clone()));
```

Comment:
Why are you building a new structure by cloning everything? Please prefer iterator-based construction via collect() for simplicity and zero-cost abstraction. Also you wouldn't need mutable entries